realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -66,13 +67,140 @@
         void onCall();
     }
 
-    private static class RefAndCount {
-        // The Realm instance in this thread.
-        private final ThreadLocal<BaseRealm> localRealm = new ThreadLocal<>();
+    private static abstract class ReferenceCounter {
+
         // How many references to this Realm instance in this thread.
-        private final ThreadLocal<Integer> localCount = new ThreadLocal<>();
+        protected final ThreadLocal<Integer> localCount = new ThreadLocal<>();
         // How many threads have instances refer to this configuration.
-        private int globalCount = 0;
+        protected int globalCount = 0;
```

Comment:
This is accessed potentially from various Threads, make it `volatile` or use atomic (`AtomicInteger`).